[Jeremy Tan] iP

koonweee opened the pull request.

Commit:
Add Gradle support

koonweee committed:
Greet, Echo, Exit

koonweee committed:
Add, List

koonweee committed:
Mark as Done

koonweee committed:
ToDo, Event, Deadline

koonweee committed:
Test using script

koonweee committed:
Handle Errors

koonweee committed:
Delete

koonweee committed:
no message

koonweee committed:
Save

koonweee committed:
Dates and Times

koonweee committed:
Merge branch 'branch-Level-7'

koonweee committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

koonweee committed:
Use More OOP

koonweee committed:
Use More OOP

koonweee committed:
Add JUnit Tests

koonweee committed:
Add JUnit Tests

koonweee committed:
JavaDoc

koonweee committed:
Follow the Coding Standard

koonweee committed:
Find

koonweee committed:
Merge branch 'branch-A-JavaDoc'

koonweee committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Command.java

koonweee committed:
Merge branch 'branch-Level-9'

koonweee committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

koonweee committed:
Use Gradle

koonweee committed:
Use CheckStyle

koonweee committed:
Merge branch 'branch-A-Gradle'

koonweee committed:
Merge branch 'branch-A-CheckStyle'

# Conflicts:
#	build.gradle

koonweee committed:
GUI

koonweee committed:
Merge branch 'branch-Level-10'

daongochieu2810 commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,213 @@
+     */
+
+
+    static Command parse(String command) throws InvalidInput {
```
This could be handled by a `Parser` class

daongochieu2810 commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,213 @@
+    }
+}
+
+class FindCommand extends Command {
```
Shouldnt different classes be put into different files ?

rgabelarde replied:
Agreed, for now the code is manageable as they are not too much, but if the project expands, it may become hard to navigate.

daongochieu2810 commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,74 @@
+     * Formats given String and prints in the UI
+     * @param printStr String to be formatted and printed
+     */
+    public void printf(String printStr) {
```
I think this function name should be more expressive

daongochieu2810 commented:
Great job with the current iP progress! The code is neat and most classes have specific methods with 1 clear purpose. However there are several points that could be improved. The testing seems to be a bit sparse, you might want to add more tests as you have quite a number of methods for each class.
Good job overall!

rgabelarde commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,210 @@
+class DeleteCommand extends Command {
+    private int deleteTask;
+
+    DeleteCommand(int task) {
```
Can consider renaming the _task_ parameter as _taskIndex_ to make it clearer for users

rgabelarde commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,210 @@
+ * A delete command to delete a given task number
+ */
+class DeleteCommand extends Command {
+    private int deleteTask;
```
Could be named better, maybe _taskIndexDelete_, etc..

ChenXJ98 commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,210 @@
+class TodoCommand extends Command {
+    private String task;
+
+    TodoCommand(String toParse) {
```
i agree that it will be better to split the various classes into different files

ChenXJ98 commented on src/main/java/DialogBox.java:
```diff
@@ -0,0 +1,59 @@
+        setAlignment(Pos.TOP_LEFT);
+    }
+
+    public static DialogBox getUserDialog(String text, Image img) {
+        return new DialogBox(text, img);
+    }
+
+    public static DialogBox getDukeDialog(String text, Image img) {
+        var db = new DialogBox(text, img);
+        db.flip();
+        return db;
```
Not sure if Javadocs is needed since the code is provided.

ChenXJ98 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,83 @@
+        new Duke("data/tasks.txt").run();
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
```
Is this a placeholder for now?

ChenXJ98 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+        return Arrays.asList(rtnSplit).stream().collect(Collectors.joining("\n"));
+    }
+
+    public String find(String key) {
```
Might be better if function is renamed to finding a specific thing instead of just find.

ChenXJ98 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+            String[] split = line.split("%d%");
+            if (split.length > 3) {
+                taskList.add(Task.createTask(split[0], split[1], split[2], split[3]));
+            } else {
```
Can add an additional else statement to capture everything else. For easier debugging.

ChenXJ98 reviewed (commented):
Overall really great! I love that your code is very neat and easy to understand. Just some nit picks actually. :D

koonweee committed:
Fix bugs with text output

koonweee committed:
Fix additional bugs within TaskList and added Jar file

koonweee committed:
Use Assertions: Add assertion for storage file type

koonweee committed:
Improve Code Quality:
Check for code quality violations and standardized formatting throughout.
Refactor Commands into separate class files

koonweee committed:
Merge pull request #1 from koonweee/A-Assertions

Use Assertions: Add assertion for storage file type

koonweee committed:
Merge pull request #2 from koonweee/A-CodeQuality

Improve Code Quality

koonweee committed:
Flexible data source
Add prompt to load data file using command "load"

koonweee committed:
Flexible data source
Add prompt to select data file when "load" command given

koonweee committed:
Merge branch 'branch-C-FlexibleDataSource'

koonweee committed:
Update GUI
New profile pictures
Minimal GUI

koonweee committed:
no message

koonweee committed:
Add Ui Screenshot to Docs

koonweee committed:
Add a User Guide

koonweee committed:
Set theme jekyll-theme-minimal

koonweee committed:
Fix markdown formatting

koonweee committed:
Revert "Fix markdown formatting"

This reverts commit cfa9af54dc5f4415824ba19f4141518048400602.

koonweee committed:
Merge branch 'master' of https://github.com/koonweee/ip

koonweee committed:
Fixed Markdown formatting

koonweee committed:
Fix Markdown formatting

koonweee committed:
Fix Storage.java

koonweee committed:
Fix input validation and checkstyle issues

Valid input dates via regex. Restrict file loading to .txt files only. Add Javadoc for more classes and methods.

koonweee committed:
Update user guide

koonweee committed:
Fix checkstyle issues
